src/async.c

```diff
@@ -1,7 +1,15 @@
 #include <assert.h>
+
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__DragonFly__)
+#include <stdlib.h>
```

Comment:
Since this is included in both branches, what do you think about moving it up to just under `assert.h`?